dynamicPermissions.ps1

```diff
@@ -0,0 +1,464 @@
+            VerboseErrorMessage = $null
+            AuditErrorMessage   = $null
+        }
+
+        try {
+            $errorMessage.VerboseErrorMessage = $ErrorObject.ErrorDetails.Message
+            $errorMessage.AuditErrorMessage = Resolve-ZenyaErrorMessage $ErrorObject.ErrorDetails.Message
+        }
+        catch {
+            Write-Verbose "Error resolving Zenya error message, using default Powershell error message"
+        }
+
+        # If error message empty, fall back on $ex.Exception.Message
+        if ([String]::IsNullOrEmpty($errorMessage.VerboseErrorMessage)) {
+            $errorMessage.VerboseErrorMessage = $ErrorObject.Exception.Message
+        }
+        if ([String]::IsNullOrEmpty($errorMessage.AuditErrorMessage)) {
+            $errorMessage.AuditErrorMessage = $ErrorObject.Exception.Message
+        }
+
+        Write-Output $errorMessage
+    }
+}
+
+function New-AuthorizationHeaders {
+    [CmdletBinding()]
+    [OutputType([System.Collections.Generic.Dictionary[[String], [String]]])]
+    param(
+        [parameter(Mandatory)]
+        [string]
+        $ClientId,
+
+        [parameter(Mandatory)]
+        [string]
+        $ClientSecret
+    )
+    try {
+        Write-Verbose 'Creating Access Token'
+
+        $authorizationurl = "$baseUrl/oauth/token"
+        $authorizationbody = @{
+            "grant_type"                = 'client_credentials'
+            "client_id"                 = $ClientId
+            "client_secret"             = $ClientSecret
+            "token_expiration_disabled" = $false
+        } | ConvertTo-Json -Depth 10
+        $AccessToken = Invoke-RestMethod -uri $authorizationurl -body $authorizationbody -Method Post -ContentType "application/json"
+
+        Write-Verbose 'Adding Authorization headers'
+        $headers = [System.Collections.Generic.Dictionary[[String], [String]]]::new()
+        [void]$headers.Add('Authorization', "$($AccessToken.token_type) $($AccessToken.access_token)")
+        [void]$headers.Add('Accept', 'application/json')
+        [void]$headers.Add('Content-Type', 'application/json')
+        Write-Output $headers
+    }
+    catch {
+        throw $_
+    }
+}
+#endregion functions
+
+try {
+    # Create authorization headers
+    try {
+        $headers = New-AuthorizationHeaders -ClientId $clientId -ClientSecret $clientSecret
+    }
+    catch {
+        $ex = $PSItem
+        $errorMessage = Get-ErrorMessage -ErrorObject $ex
+
+        Write-Verbose "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
+
+        $outputContext.AuditLogs.Add([PSCustomObject]@{
+                # Action  = "" # Optional
+                Message = "Error creating authorization headers. Error Message: $($errorMessage.AuditErrorMessage)"
+                IsError = $true
+            })
+
+        # Throw terminal error
+        throw
+    }
+
+    # Get groups
```

Comment:
we should get the groups using a paged construction. See: https://identitymanagement.services.iprova.nl/swagger-ui/#!/scim/GetGroupsRequest
When there are a lot of groups, this could be a potential issue.

Reply:
Agreed, paging has been added.